pageserver/src/tenant/timeline.rs

```diff
@@ -4344,30 +4345,39 @@ impl Timeline {
                 next_start_key: img_range.end,
             });
         }
-        let has_keys = !data.is_empty();
+        let mut has_keys = false;
```

Comment:
I think this is a bit weird naming, because it now means `wrote_any_images`. Checking if that `v.is_empty()` has any docs about it suggesting a tombstone..

Reply:
Ah, v is `bytes::Bytes`. Additionally I think we should add back that comment like:

```
if v.is_empty() {
    // the key has been deleted, it does not need an image
    continue;
}
```

Reply:
Why was this resolved without comments?

Reply:
Aha, I see you added a comment within. Could you at least explain to me what is the rationale of "has_keys" vs. "wrote_any_images". I think the latter would better capture the intent (wrote any images to the writer).